workflows_core/engine/small_batch_stable_engine.py

```diff
@@ -0,0 +1,144 @@
+"""
+    Stable Engine Pseudo-algorithm-
+        1. Downloads about 1000 documents.
+        2. Processes it in much smaller chunks.
+        3. Upserts all 1000 documents.
+        4. Repeat until dataset has finished looping
+
+    We download a large chunk and upsert large chunks to avoid hammering
+    our servers.
+
+"""
+import logging
+import traceback
+
+from workflows_core.operator.abstract_operator import AbstractOperator
+from workflows_core.dataset.dataset import Dataset
+from workflows_core.engine.abstract_engine import AbstractEngine
+from workflows_core.utils.document_list import DocumentList
+from tqdm.auto import tqdm
+
+logger = logging.getLogger(__file__)
+
+
+class SmallBatchStableEngine(AbstractEngine):
+    def __init__(
+        self,
+        dataset: Dataset,
+        operator: AbstractOperator,
+        pull_chunksize: int = 5,
+        transform_threshold: int = 1000,
```

Comment:
`transform_threshold` is the minimum amount of documents that are required before it starts running the `transform` function whereas the `transform_chunksize` is the number of documents each `transform` operation will take.